pqarrow/arrow.go

```diff
@@ -25,7 +25,7 @@ func ParquetRowGroupToArrowRecord(
 	projections []logicalplan.ColumnMatcher,
 	filterExpr logicalplan.Expr,
 	distinctColumns []logicalplan.ColumnMatcher,
-) (arrow.Record, error) {
+) (arrow.Record, *arrow.Schema, error) {
```

Comment:
Am I missing something or can we already read the schema from the `arrow.Record`? If so I think we should just return the `arrow.Record` by itself and let the caller decide what to do.